lib/parser.js

```diff
@@ -0,0 +1,101 @@
+'use strict'
+const path = require('path')
+
+module.exports = class Parser {
+
+  static commandParser (opts) {
+    const command = opts.command
+    if (!command) return null
+    if (Array.isArray(command)) {
+      return command.join(' ')
+    }
+    return command
+  }
+
+  static buildDockerfilePathParser (opts) {
+    const build = opts.build
+    const warnings = opts.warnings
+    let dockerFileName = 'Dockerfile'
+    if (!build) return null
+
+    if (build.args) {
+      // TODO: Maybe pass these to the ENVs?
+      warnings.push({
+        message: 'The `args` argument is not supported for builds. These args will not be passed to the build.',
+        args: build.args
+      })
+    }
+
+    let buildPath
+    if (typeof build === 'string') {
+      buildPath = build
+    } else {
+      buildPath = build.context
+      if (build.dockerfile) {
+        dockerFileName = build.dockerfile
+      }
+    }
+    return path.resolve('/', buildPath, dockerFileName)
+  }
+
+  static portsParser (opts) {
+    const ports = opts.ports
+    // const warnings = opts.warnings
+    if (!ports) return []
+    return ports.map(port => {
+      if (port.match(/[0-9]*/)) {
+        return parseInt(port, 10)
+      }
+    })
+  }
+
+  static envParser (opts) {
+    const env = opts.env
+    return env
+  }
+
+  static filesParser (opts) {
+    const image = opts.image
+    if (image) {
+      return {
+        '/Dockerfile': {
+          body:
+`# Image automatically created from docker-compose file
+FROM ${image}`
+        }
+      }
+    }
+    return null
+  }
+
+  static nameParser (opts) {
+    const serviceName = opts.serviceName
+    // const mainServiceName = opts.mainServiceName
+    const repositoryName = opts.repositoryName
+    return `${repositoryName}-${serviceName}`.replace(/\W/g, '-')
+  }
+
+  static envReplacementParser (opts) {
+    const env = opts.env
+    const hostnames = opts.hostnames
+    const links = opts.links
```

Comment:
Not really because of this part of the code: 

```
  const links = services[serviceKey].links || []
        const dependsOn = services[serviceKey].depends_on || []
        return {
          metadata: {
            name: serviceKey,
            isMain: serviceKey === mainServiceName,
            links: links.concat(dependsOn)
```
I'd almost want to have a way to proof that we're always passing this, but not sure if there's a way to do that other than unit tests.